### PR TITLE
Add nxos_l2_interfaces resource module

### PR DESCRIPTION
**Pull request by trishnaguha**
Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

modules/network/nxos/nxos_l2_interfaces.py

### CODE REVIEW

File: lib/ansible/modules/network/nxos/nxos_l2_interfaces.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+#   ipv6 address auto-config
+
+- name: Merge provided configuration with device configuration.
+  nxos_l2_interfaces:
+    config:
+      - name: Ethernet1/1
+        mode: trunk
+        trunk:  
+	  native_vlan: 10
+	  allowed_vlans: 2,4,15
```
Comment (danielmellado) [inline lib/ansible/modules/network/nxos/nxos_l2_interfaces.py]: There's some mixed spaces and tabs here, please remove the tabs and use spaces only.

File: lib/ansible/modules/network/nxos/nxos_l2_interfaces.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+      - name: Ethernet1/2
+        access:
+	  vlan: 30
```
Comment (danielmellado) [inline lib/ansible/modules/network/nxos/nxos_l2_interfaces.py]: same here

File: lib/ansible/modules/network/nxos/nxos_l2_interfaces.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+#   ip address dhcp
+#   ipv6 address auto-config
+
+- name: Replace device configuration of specified L2 interfaces with provided configuration.
+  nxos_l2_interfaces:
+    config:
+      - name: Ethernet1/1
+        mode: trunk
+        trunk:
+	  native_vlan: 20
```
Comment (danielmellado) [inline lib/ansible/modules/network/nxos/nxos_l2_interfaces.py]: ditto

File: lib/ansible/modules/network/nxos/nxos_l2_interfaces.py [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+	  trunk_vlans: 5-10, 15
```
Comment (danielmellado) [inline lib/ansible/modules/network/nxos/nxos_l2_interfaces.py]: ditto

File: lib/ansible/modules/network/nxos/nxos_l2_interfaces.py [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+# interface mgmt0
+#   ip address dhcp
+#   ipv6 address auto-config
+
+- name: Override device configuration of all L2 interfaces on device with provided configuration.
+  nxos_l2_interfaces:
+    config:
+      - name: Ethernet1/2
+        access:
+	  vlan: 30
```
Comment (danielmellado) [inline lib/ansible/modules/network/nxos/nxos_l2_interfaces.py]: ditto

**Review by danielmellado — CHANGES_REQUESTED** [review]
Some minor typos there, thanks Trishna!

**Review by samccann — COMMENTED** [review]
no other docs comments other than the spacing comments from the other reviewer

**Comment by trishnaguha** [comment]
@danielmellado fixed the spacing.

**Review by danielmellado — APPROVED** [review]
LGTM now, thanks trishna!

**Pull request merged**